populator-machinery/controller.go

```diff
@@ -108,11 +114,36 @@ type controller struct {
 	recorder             record.EventRecorder
 	referenceGrantLister referenceGrantv1beta1.ReferenceGrantLister
 	referenceGrantSynced cache.InformerSynced
+	// Boolean flag which determines whether or not to use cloud provider specific data populate functions
+	useProviderImpl bool
+	// Data population function, invoked when the useProviderImpl variable is set to true
+	populate func(context.Context, *PopulatorParams) error
+	// Data population completeness check function, return true when data transfer gets completed.
+	// Invoked when the useProviderImpl variable is set to true
+	populateComplete func(context.Context, *PopulatorParams) (bool, error)
+}
+
+type PopulatorParams struct {
+	KubeClient   kubernetes.Interface
+	StorageClass *storagev1.StorageClass
```

Comment:
We need a ScPrime as well?

Reply:
We don't have a use case with ScPrime for now, the original StorageClass is good enough